src/librbd/internal.cc

```diff
@@ -3513,4 +3513,143 @@ int mirror_image_disable_internal(ImageCtx *ictx, bool force,
       ictx->perfcounter->inc(l_librbd_readahead_bytes, readahead_length);
     }
   }
+
+  // Consistency groups functions
+
+  int create_cg(librados::IoCtx& io_ctx, const char *cg_name)
+  {
+    ostringstream bid_ss;
+    uint32_t extra;
+    int remove_r;
+    string id, id_cg, header_oid;
+
+    CephContext *cct = (CephContext *)io_ctx.cct();
+
+    Rados rados(io_ctx);
+    uint64_t bid = rados.get_instance_id();
+
+    id_cg = util::id_cg_name(cg_name);
```

Comment:
I have to be honest, I never saw the purpose of the RBD image v2 id object.  We should add the CG to the directory by providing a name and "unique" id. If the name or id is in-use, the exec call will fail with -EEXISTS/-ESTALE.  Once we have it recorded in the directory, so long as we have "CG name"->"CG id" and "CG id"->"CG name" mappings (see RBD image directory), we don't need the "cg_id.<CG name>" object and on rename we only have to update the directory and not the directory and the id object.